pandas/tests/indexing/multiindex/test_slice.py

```diff
@@ -492,6 +492,29 @@ def test_loc_axis_arguments(self):
         with pytest.raises(ValueError):
             df.loc(axis="foo")[:, :, ["C1", "C3"]]
 
+    def test_loc_axis_single_level_indexer(self):
+
+        # test single level indexing on Multindex column dataframe
+        df1 = pd.DataFrame(
+            np.arange(27).reshape(3, 9),
+            columns=pd.MultiIndex.from_product(
+                [["a1", "a2", "a3"], ["b1", "b2", "b3"]]
+            ),
+        )
+        result = df1.loc(axis=1)["a1":"a2"]
+        expected = df1.iloc[:, :-3]
+        tm.assert_frame_equal(result, expected)
+        result = df1.loc(axis=1)["a1"]
+        expected = df1.iloc[:, :3]
+        expected.columns = ["b1", "b2", "b3"]
+        tm.assert_frame_equal(result, expected)
```

Comment:
Could you make 3 different functions for these tests?

Reply:
@mroeschke Is it ok if I split it into 2 functions - one for Multindex column df and one for simple df --- and for the Multiindex bit I can comment on the two different tests - i.e one is slicing on multi columns on single level and other is slicing on single col. Otherwise, having concise names for the functions is tricky.

Something like this:
```
        # GH29519
        # test with single level multiple columns slice
        df1 = pd.DataFrame(
            np.arange(27).reshape(3, 9),
            columns=pd.MultiIndex.from_product(
                [["a1", "a2", "a3"], ["b1", "b2", "b3"]]
            ),
        )
        result = df1.loc(axis=1)["a1":"a2"]
        expected = df1.iloc[:, :-3]

        tm.assert_frame_equal(result, expected)

        # GH29519
        # test with single level single column slice
        result = df1.loc(axis=1)["a1"]
        expected = df1.iloc[:, :3]
        expected.columns = ["b1", "b2", "b3"]

        tm.assert_frame_equal(result, expected)


```

Reply:
Verbose function names are fine. Helps isolate the failing test as separate methods.

Reply:
Ok just pushed with changes as you originally suggested